src/Writer/Factory/WriterFactory.php

```diff
@@ -0,0 +1,126 @@
+     *
+     * @param null|array
+     */
+    private $creationOptions;
+
+    /**
+     * @param null|array|Traversable $creationOptions
+     * @throws InvalidServiceException if $creationOptions cannot be coerced to
+     *     an array.
+     */
+    public function __construct($creationOptions = null)
+    {
+        if (null === $creationOptions) {
+            return;
+        }
+
+        $this->setCreationOptions($creationOptions);
+    }
+
+    /**
+     * Create an instance of the requested class name.
+     *
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @param null|array $options
+     * @return object
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        if (isset($options['filter_manager']) && is_string($options['filter_manager'])) {
+            $options['filter_manager'] = $container->get($options['filter_manager']);
+        } elseif (!isset($options['filter_manager']) && $container->has('LogFilterManager')) {
+            $options['filter_manager'] = $container->get('LogFilterManager');
+        }
+
+        if (isset($options['formatter_manager']) && is_string($options['formatter_manager'])) {
+            $options['formatter_manager'] = $container->get($options['formatter_manager']);
+        } elseif (!isset($options['formatter_manager']) && $container->has('LogFormatterManager')) {
+            $options['formatter_manager'] = $container->get('LogFormatterManager');
+        }
+
+        return (null === $options) ? new $requestedName : new $requestedName($options);
+    }
+
+    /**
+     * Create an instance of the named service.
+     *
+     * First, it checks if `$canonicalName` resolves to a class, and, if so, uses
+     * that value to proxy to `__invoke()`.
+     *
+     * Next, if `$requestedName` is non-empty and resolves to a class, this
+     * method uses that value to proxy to `__invoke()`.
+     *
+     * Finally, if the above each fail, it raises an exception.
+     *
+     * The approach above is performed as version 2 has two distinct behaviors
+     * under which factories are invoked:
+     *
+     * - If an alias was used, $canonicalName is the resolved name, and
+     *   $requestedName is the service name requested, in which case $canonicalName
+     *   is likely the qualified class name;
+     * - Otherwise, $canonicalName is the normalized name, and $requestedName
+     *   is the original service name requested (typically the qualified class name).
+     *
+     * @param ServiceLocatorInterface $serviceLocator
+     * @param null|string $canonicalName
+     * @param null|string $requestedName
+     * @return object
+     * @throws InvalidServiceException
+     */
+    public function createService(ServiceLocatorInterface $serviceLocator, $canonicalName = null, $requestedName = null)
+    {
+        if (class_exists($canonicalName)) {
```

Comment:
Should test that `$canonicalName` is a string first, as you do with `$requestedName`.